pushapkscript/test/test_googleplay.py

```diff
@@ -38,80 +43,111 @@ def setUp(self):
+                False,
+                True,
+            )
+            args, _ = mock_push_apk.call_args
+            google_play_strings = args[5]
+            assert isinstance(google_play_strings, NoGooglePlayStrings)
+
+    def test_publish_allows_rollout_percentage(self, mock_push_apk):
         self.context.task['payload']['google_play_track'] = 'rollout'
         self.context.task['payload']['rollout_percentage'] = 10
-        self.assertEqual(craft_push_apk_config(self.context, self.apks), {
-            '*args': ['/path/to/arm_v15.apk', '/path/to/x86.apk'],
-            'credentials': '/path/to/release.p12',
-            'commit': False,
-            'no_gp_string_update': True,
-            'rollout_percentage': 10,
-            'service_account': 'release_account',
-            'track': 'rollout',
-        })
-
-    def test_craft_push_config_allows_to_contact_google_play_or_not(self):
-        config = craft_push_apk_config(self.context, self.apks)
-        self.assertNotIn('do_not_contact_google_play', config)
-
-        self.context.config['do_not_contact_google_play'] = True
-        config = craft_push_apk_config(self.context, self.apks)
-        self.assertTrue(config['do_not_contact_google_play'])
+        publish_to_googleplay(self.context, self.apks)
+        mock_push_apk.assert_called_once_with(
+            ANY,
+            ANY,
+            ANY,
+            'rollout',
+            10,
+            ANY,
+            ANY,
+            ANY,
+        )
 
-    def test_craft_config_instance_config_file_allows_to_contact_google_play_or_not(self):
-        config = craft_push_apk_config(self.context, self.apks)
-        self.assertNotIn('do_not_contact_google_play', config)
+    def test_publish_allows_to_contact_google_play_or_not(self, mock_push_apk):
+        publish_to_googleplay(self.context, self.apks)
+        mock_push_apk.assert_called_with(
+            ANY,
+            ANY,
+            ANY,
+            ANY,
+            ANY,
+            ANY,
+            ANY,
+            True,
+        )
 
         self.context.config['do_not_contact_google_play'] = True
-        config = craft_push_apk_config(self.context, self.apks)
-        self.assertTrue(config['do_not_contact_google_play'])
+        publish_to_googleplay(self.context, self.apks)
+        mock_push_apk.assert_called_with(
+            ANY,
+            ANY,
+            ANY,
+            ANY,
+            ANY,
+            ANY,
+            ANY,
+            False,
+        )
 
-    def test_craft_push_config_allows_committing_apks(self):
-        self.context.task['scopes'] = ['project:releng:googleplay:aurora']
+    def test_publish_allows_committing_apks(self, mock_push_apk):
         self.context.task['payload']['commit'] = True
-        config = craft_push_apk_config(self.context, self.apks)
-        self.assertTrue(config['commit'])
+        publish_to_googleplay(self.context, self.apks)
+
+        mock_push_apk.assert_called_with(
```

Comment:
I could peel argument at index `[i]` out and verify that it's `True`.
I'm not sure if that's more or less clean than this `assert_called_with` with a bunch of `ANY`.
Let me know :)